browser/ui/webui/instant_service_message_handler.cc

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2019 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/instant_service_message_handler.h"
+
+#include <string>
+#include <memory>
+#include <utility>
+
+#include "base/i18n/rtl.h"
+#include "base/strings/utf_string_conversions.h"
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/browser/ui/webui/brave_new_tab_ui.h"
+#include "brave/browser/ntp_background_images/view_counter_service_factory.h"
+#include "brave/common/pref_names.h"
+#include "brave/components/ntp_background_images/browser/features.h"
+#include "brave/components/ntp_background_images/browser/ntp_background_images_data.h"
+#include "brave/components/ntp_background_images/browser/view_counter_service.h"
+#include "brave/components/ntp_background_images/common/pref_names.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/search/instant_service.h"
+#include "content/public/browser/web_ui_data_source.h"
+
+using ntp_background_images::features::kBraveNTPBrandedWallpaper;
```

Comment:
I don't know if this and the other includes above are needed @simonhong - I just copied from the other handler